modules/mobile/app/services/mobile/v2/appointments/proxy.rb

```diff
@@ -19,8 +19,12 @@ def get_appointments(start_date:, end_date:, include_pending:, pagination_params
           # or the upstream serice does not use them.
           response = vaos_v2_appointments_service.get_appointments(start_date, end_date, statuses.join(','),
                                                                    pagination_params)
+          appointments = response[:data]
 
-          appointments = merge_clinic_facility_address(response[:data])
+          # cancelled appointments can also be pending/proposed appointments but under a different status
+          appointments = appointments.select { |appt| appt[:requested_periods].blank? } unless include_pending
```

Comment:
I think this should be happening in the controller for two reasons:
1) we already filter out pending (non-cancelled) in the controller so it feels weird to do different parts of that filter in two places
2) the reason we do that in the controller is that all filtering should happen after caching, which happens in the controller.

Reply:
Yeah, It's definitely awkward having it in two places but this way this will minimize the amount of calls being made to other upstream backfill services. As for the caching, I was thinking about this more and probably should have mentioned it in the PR notes. while the cache would be inconsistent if you called the endpoint with pending included and then right after called it again with pending not included, this won't ever happen in the app, only users that never update the app will call it without pending included so the cache will be consistent for them except for the short time after they update the app. This still isn't ideal since can potentially break the caching when using it outside the app so I'm not opposed to still keeping this in the controller.

Reply:
sounds good. comment updated.